### PR TITLE
BTM 621 UI tests

### PR DESCRIPTION
**Pull request by NKannan14**
This PR introduces new UI tests that are structured using pageObjects,spec test files test data, utils and helper files. The tests validate the UI data by comparing it with the test data from JSON file.

Due to module compatibility issue that prevents importing files without `.js` extension so unable to use existing s3 helper and client files. So created separate s3 helper and client files. The files handle the uploading of test data to S3 bucket.
But if there are any ways to avoid the code repeatability please let me know.

Currently tests designed to run specifically in Chrome, Configuration to run tests on other browsers will be added in future tickets.

### CODE REVIEW

File: ui-tests/clients/clients.ts
[the remark this answers is not in the document]
Comment (NKannan14) [reply ui-tests/clients/clients.ts:1]: Great! Thanks Gareth

[the remark this answers is not in the document]
Comment (NKannan14) [reply ui-tests/clients/clients.ts:1]: @gdsgj  is it okay to merge refactor branch into this branch?

File: ui-tests/specs/invoicesList.spec.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  getTestDataFilePath,
+  getUniqueVendorNamesFromJson,
+} from "../utils/extractTestDatajson";
+
+/* UI tests for Invoices List Page. It verifies that the correct vendor name is displayed on the page.
+It includes tests to ensure that the unique invoice count for each vendor matches the count obtained from the UI */
+
+describe("InvoicesList Page", () => {
+  enum VendorNameContractIdMap {
+    "Vendor One" = 1,
```
[the remark this answers is not in the document]
Comment (NKannan14) [reply ui-tests/specs/invoicesList.spec.ts:14]: Thanks, I will change it to retrieve the contract id from the config

**Review by Si3rr4wow — APPROVED** [review]

File: ui-tests/specs/invoicesList.spec.ts [hunk 2]
```diff
@@ -1,48 +1,44 @@
...
 
   vendorsNameFromJson.forEach((vendor) => {
     it(`should display the correct vendor name for ${vendor}`, async () => {
-      const vendorContractId =
-        VendorNameContractIdMap[vendor as keyof typeof VendorNameContractIdMap];
-      await InvoicesListPage.open(`contracts/${vendorContractId}/invoices`);
-      await waitForPageLoad();
-      const uiVendorName = await InvoicesListPage.getPageSubHeadingText();
-      expect(uiVendorName).toContain(`${vendor}`);
+      const vendorIds = getUniqueVendorIdsFromJson(vendor);
```
[the remark this answers is not in the document]
Comment (NKannan14) [reply ui-tests/specs/invoicesList.spec.ts:20]: yes, for staging will  put the testData extract file in config repo as it has vendor details, when we deploy the site in staging will add the file in config repo and test it. The current extract test data json file in the repo for only dev and build.

**Pull request merged**